### PR TITLE
Adding Arm-Neon implementation.

### PR DESCRIPTION
**Pull request by RealTimeChris**
[Adding Arm-Neon Implementation]

## Code change checklist

- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html).
- [x] I tested that my change works before raising the PR.
- [x] I have ensured that I did not break any existing API calls.
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight.

### CODE REVIEW

File: cmake/DetectArchitecture.cmake [hunk 1]
```diff
@@ -43,7 +50,7 @@ set(AVX_TYPE "AVX0" PARENT_SCOPE)
 set(AVX_FLAGS "" PARENT_SCOPE)
 
 # This is only supported on x86/x64, it is completely skipped and forced to T_fallback anywhere else
-if ((${CMAKE_SYSTEM_PROCESSOR} MATCHES "x86_64") OR (${CMAKE_SYSTEM_PROCESSOR} MATCHES "i386") OR (${CMAKE_SYSTEM_PROCESSOR} MATCHES "AMD64"))
+if ((${CMAKE_SYSTEM_PROCESSOR} MATCHES "x86_64") OR (${CMAKE_SYSTEM_PROCESSOR} MATCHES "i386") OR (${CMAKE_SYSTEM_PROCESSOR} MATCHES "AMD64") OR (${CMAKE_HOST_SYSTEM_PROCESSOR} MATCHES "arm64"))
```
Comment (braindigitalis) [inline cmake/DetectArchitecture.cmake:53]: what about arm7hf? arm7hf can have neon

**Review by braindigitalis — CHANGES_REQUESTED** [review]

**Review by RealTimeChris — COMMENTED** [review]
Looks good to me.

File: cmake/DetectArchitecture.cmake (hunk 1 above)
Comment (RealTimeChris) [reply cmake/DetectArchitecture.cmake:53]: https://stackoverflow.com/a/70498851
According to this, these are the possible values for this value - how do we detect if it's arm7hf?

Comment (braindigitalis) [reply cmake/DetectArchitecture.cmake:53]:
```
brain@raspberrypi:~ $ uname -m
armv7l
```

Looks like `armv7l` is what you need to look for

**Review by Mishura4 — APPROVED** [review]

**Comment by braindigitalis** [comment]
the correct actions havent run to test this compiles. Can we please make a change to this which will trigger the correct CI

**Review by braindigitalis — APPROVED** [review]

**Pull request merged**